pages/a/1k1-azure-post.html

```diff
@@ -0,0 +1,239 @@
+for scheduling Crate instances and [Consul](https://www.consul.io/) for node discovery in a Swarm
+cluster. The solution has its pros and cons. Scheduling Crate nodes, and being able to change
+settings in the Crate cluster with minimal effort (in order to repeat certain experiments, by any
+means), is the strongest side of the chosen approach. On the other hand, it adds complexity to the
+overhead in the Crate cluster setup. For example, Consul and Swarm cluster must be deployed first.
+Moreover, the elaborate setup results in higher network load and additional connections introduced
+by backend clusters.
+
+### Plain Crate setup
+
+For this reason, we wanted to be certain that the high network traffic was not due to Crate, so
+our next plan was to go back to basics and eliminate other potential sources of noise. Therefore,
+we decided to use the Crate Debian package to setup the cluster and conduct test using as simple as
+possible approach for spinning up the cluster.
+
+
+## Azure Setup
+
+We aimed to create a 1001 node Crate cluster ready for data import and manipulation. Setting up such
+a cluster on any kind of infrastructure requires planning and whilst cloud hosts simplify this
+process, understanding how to translate your requirements into their paradigms requires research
+and experimentation.
+
+Azure has several concepts that we needed to understand and make use of to get our cluster started:
+
+- **Resource Group**: A 'container' that holds related resources of all types for an application.
+- **Storage Account**: A data storage solution allocated to you that can contain Blob, Table, Queue,
+    and File data. Can be stored on SSDs or HDDs.
+- **Virtual Network (vnet)**: A representation of a network in the cloud, where you can define DHCP
+    address blocks, DNS settings, security policies, and routing.
+- **Subnet**: Further segmentation for virtual networks.
+- **Virtual Machine (VM)**: Represents a particular operating system or application stack.
+- **Network Interface (NIC)**: Represents a network interface that can be associated to one or more
+    virtual machines (VM).
+- **Network Security Group (NSG)**: Contains a list of rules that allow or deny network traffic to
+    your VMs in a Virtual Network. Can be associated with subnets or individual VM instances within
+    that subnet.
+
+### Limits
+
+These are some platform specific limitations that influenced our architecture decisions:
+
+- 8000 CPU cores
+- 10000 VMs
+- 100 VMs in single availability set
+- 800 resources in single resource group
+- 4096 IP addresses in a virtual network (VNet)
+
+### ARM Templates
+
+[Azure Templates]
+(https://azure.microsoft.com/en-us/documentation/articles/resource-group-authoring-templates/), a
+json formatted file that provisions all the resources and commands to be run for your application
+in a single operation. These templates are used in conjunction with the Azure command line tool to
+trigger a complex cluster quickly.
```

Comment:
Maybe something about this in-string-code-constructions. ARMs are able to handle.